File: LanguageTutorials/DesignPatterns/src/main/java/org/bytescale/patterns/observer/CurrentConditionsDisplay.java
package org.bytescale.patterns.observer;

import java.util.Observable;

public class CurrentConditionsDisplay implements java.util.Observer, DisplayElement {
    private Observable observable;
    private float temperature;
    private float humidity;

    public CurrentConditionsDisplay(Observable observable) {
        this.observable = observable;
        this.observable.addObserver(this);
    }

    public void display() {
        System.out.println("Current conditions: " + temperature + "F degrees and " + humidity + "% humidity.");
    }

    public void update(Observable observable, Object o) {
        if (observable instanceof  WeatherData) {
            WeatherData weatherData = (WeatherData)observable;
            this.temperature = weatherData.getTemperature();
            this.humidity = weatherData.getHumidity();
            display();
        }
    }
}

File: LanguageTutorials/CSDesignPatterns/Org.Bytescale.Patterns/Observer/CurrentConditionsDisplay.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Org.Bytescale.Patterns.Observer
{
    public class CurrentConditionsDisplay : IObserver<WeatherData>
    {
        public void display(float humidity, float temperature, float pressure) {
            Console.WriteLine("Current conditions: " + temperature + "F degrees, " + humidity + "% humidity, and " + pressure + " barometric pressure.");
        }

        public void OnCompleted()
        {
            Console.WriteLine("All stations have reported.");
        }

        public void OnError(Exception error)
        {
            Console.WriteLine("Error reporting weather: " + error.Message);
        }

        public void OnNext(WeatherData weatherData)
        {
            display(weatherData.Humidity, weatherData.Temperature, weatherData.Pressure);
        }
    }
}
